File: leetcode/0007-reverse-integer/solution.py
class Solution(object):
    def reverse(self, x):
        s=str(x)
        dest=''
        if s[0]=='-':
            for n in range(len(s)-1,0,-1):
                dest=dest+s[n]
            ans=int(dest)

            if ans< -2**31 or ans> 2**31 -1:
                return 0
            else:
                return ans*-1
            
        else:
            for n in range(len(s)-1,-1,-1):
                dest=dest+s[n]
            ans=int(dest)
            if ans< -2**31 or ans> 2**31 -1:
                return 0
            else:
                return ans


        """
        :type x: int
        :rtype: int
        """
        

File: leetcode/0007-reverse-integer/solution.cpp
#include <limits>
class Solution {
public:
    int reverse(int x) {
        long long res=0;
        if(x==0 || x> pow(2,31)-1 || x< -pow(2,31)){
            return 0;
        }
        
        vector<int> digits;
        while(x!=0){
            digits.push_back(x%10);
            x=x/10;
        }
        for(int i:digits){
            res=res*10;
            res=res+i;
        }
        if (res>INT_MAX || res<INT_MIN){
            return 0;
        }
        return (int)res;
    }
};
